urlextract/urlextract_core.py

```diff
@@ -695,14 +695,17 @@ def _split_markdown(text_url, tld_pos):
             return text_url[left_bracket_pos+1:middle_pos]
         return text_url
 
-    def gen_urls(self, text, check_dns=False):
+    def gen_urls(self, text, check_dns=False, get_indices=False):
         """
         Creates generator over found URLs in given text.
 
         :param str text: text where we want to find URLs
         :param bool check_dns: filter results to valid domains
+        :param bool get_indices: whether to return beginning and
+            ending indices as (<url>, (idx_begin, idx_end))
         :yields: URL found in text or empty string if no found
-        :rtype: str
+        :rtype: - str if not get_indices
```

Comment:
Could you use rtype  just for specifying type and update yield for description 
```
:yields: URL or URL with indices found in text or empty string if nothing was found
:rtype: str|tuple(str, tuple(int, int)) 
```

If you write rtype like this PyCharm and other tools can work with it. (shows warning when yo do wrong operations and so on)

Thank you!